### DISCUSSION TITLE
Local Replica fails with message: "Local Replica Scheduling Failure Error Message: tags not fulfilled"

### DISCUSSION THREAD

**Discussion by urbaman** [longhorn/longhorn#7312] [Q&A] [answered]
Hi,

I have a strage behivour on some pods.

I see 4 replicas instead of 3, on of wich is "a ghost":

![image](https://github.com/longhorn/longhorn/assets/26753344/dde064ea-a7b7-4503-a5e2-9d8f30a936f1)

Replicas are on nodes 1,2,4
The pod is scheduled (and running) on node 5 (where there's no proper replica.

The pod is properly running, as if using a proper replica.

How to properly debug and solve this problem?

Thank you.

**Comment by mantissahz** [comment]
Could you set the setting `log-level` to `Debug` or `Trace` and then delete the extra replica to check if it would still replenish an extra replica? If yes, could you generate the support bundle for the investigation?

**Comment by urbaman** [comment]
Hi @mantissahz 

How to see if the log-level has been deployed?

- Yes, deleting the ghost replica gets a new one created, see no log on the dashboard (do see the support bundle creation logs)
- Longhorn version 1.5.3 on Kubeadm 1.28.4
- How to share a big support bundle?

Thank you.

**Reply by mantissahz** [reply]
> How to see if the log-level has been deployed?

Just modify the setting `log-level` and you should be able to see `[Debug]` logs in the longhorn-manager or other Longhorn pods.

> How to share a big support bundle?

Could you send it to longhorn-support-bundle@suse.com?
Thanks.

**Comment by urbaman** [comment]
Update:

I actually have 3 workloads with the "Local Replica Scheduling Failure Error Message: tags not fulfilled" error:

1 pod with just a scheduling failure flag, no ghost replica

![image](https://github.com/longhorn/longhorn/assets/26753344/833bc539-a54d-4744-82d3-ef3656751cc5)

1 pod with the scheduling replica and just a "no data locality" flag

1 pod with both scheduling flag and data locality flag

![image](https://github.com/longhorn/longhorn/assets/26753344/49b0809d-d976-4b06-bbbd-2b8de182a1fe)

**Comment by urbaman** [comment]
Hi,

Support Bundle sent (as a Gdrive link).

PS: the third pod above does not have the same scheduling flag, but does have the ghost replica.

**Comment by urbaman** [comment]
Hi @mantissahz 

I see that the logs from the longhorn-manager of the node(s) involved says:

Pod/pvc with both no data locality and not schedulable

```
There's no available disk for replica pvc-5a3cfa97-fd0d-4fc0-b040-b004c43b0807-r-4d7062d4
Failed to schedule replica of volume with HardNodeAffinity = k8w5"
Ignoring recurring job labels on Volume pvc-5a3cfa97-fd0d-4fc0-b040-b004c43b0807 PVC teleport-kube-agent-teleport-data-teleport-kube-agent-0 due to missing source label
```

Pod/pvc with no data locality

`Failed to schedule replica of volume with HardNodeAffinity = k8w5"`

Pod/pvc not schedulable

`Ignoring recurring job labels on Volume pvc-0ea4a7ea-f28a-4879-b153-2eb4924900dc PVC uptime-kuma-pvc due to missing source label`

**Comment by urbaman** [comment]
Another insight:

If I set replicas=0 in the statefulset of a pod with no data locality, the ghost replica disappears.
If I set replocas back to 1:

- if the pod gets created on a node with a replica, everything's good.
- if the pod gets created on a node without a replica, the ghost replica appears again.

**Comment by urbaman** [comment]
@mantissahz found out that, for the "no data locality" problem, I had a label mismatch on node/disk.

I have 6 nodes, each with two disks (a virtual disk and a physical nvme disk, for a two-tier storage system).
Each node is labeled "storage", virtual disks are labeled "vdisk" and nvme disks are labeled "nvme".
I have two storage classes, one for each storage tier.

Found out that the virtual disk on node5 was labeled "nvme", as soon as I re-labeled it "vdisk" the "no data locality" problem vanished.

Also, there are no more "ghost" volumes, probably those where due to the inability to create proper volumes of tier "vdisk" on node5 on which both disks were labeled "nvme".

**I still have the "Scheduling Failure Local Replica Scheduling Failure Error Message: tags not fulfilled" problem**, can't find a way to solve it.

**Comment by urbaman** [comment, marked as answer]
Hi @mantissahz 

I finally deleted the unschedulable volumes, as the "tags not fulfilled" was probably due to the nodes/disks tag mismatch on node5 during first creation, and for some reason they didn't get the schedulable status cleared after the tag mismatch fixing.

Thank you very much for your help.

**Reply by apoca** [reply]
Hi @urbaman 

I've the same issue here and this happened when we upgrade de engine to 1.7.2

Could you please share what you did to solved it? I've read but honestly I didn't understand.

**Reply by j-karls** [reply]
I am facing the same problem. I am running on a fresh cluster with a fresh install of Longhorn 1.7.2, with default replica count set to 2, with longhorn only running on labeled nodes and data locality set to best-effort. It creates a third, unexpected ghost replica that it tries to run on the node where the workload is running (which happens to be outside of my labeled nodes). I would expect 2 replicas and no data locality, but instead I get this ghost replica.

**Reply by macrokernel** [reply]
Might be related to https://github.com/longhorn/longhorn/issues/10210?

**Reply by derekbit** [reply]
cc @PhanLe1010

**Reply by YuryHrytsuk** [reply]
Forcing your workloads to run on Nodes with Longhorn does not seem to be a solution but rather a workaround.

According to this LH KB page https://longhorn.io/kb/tip-only-use-storage-on-a-set-of-nodes/, your workloads supposed to be run on nodes without LH running (except for the case of node selectors). Following [this use case](https://longhorn.io/kb/tip-only-use-storage-on-a-set-of-nodes/#tell-longhorn-to-create-a-default-disk-on-a-specific-set-of-nodes) or [this usecase](https://longhorn.io/kb/tip-only-use-storage-on-a-set-of-nodes/#create-a-storageclass-that-select-a-specific-set-of-nodes) I would not expect to see ghost replicas and big error messages. 

This behavior seems to be a real bug

**Comment by ngrigoriev** [comment]
I just experienced this issue again when upgrading from 1.5.3 to 1.6.4. For one out of 8 volumes. And I could not delete that ghost replica using GUI - it would immediately reappear with another ID.

Last time I had this problem, I was able to delete this ghost replica.

**Comment by ngrigoriev** [comment]
...and it appears that all this was because I had a pod running on a node that did not have a disk. While I was not sure why Longhorn complained about this situation by showing several weird errors, I have decided that it was not the best. So I have configured the affinity rules to have that pod running on one of the storage-enabled nodes. And then all these errors were gone.

**Comment by derekbit** [comment]
@ngrigoriev Please provide the support bundle for investigation.
